W3ChampionsStatisticService/PersonalSettings/PersonalSettingsController.cs

```diff
@@ -11,25 +11,18 @@ namespace W3ChampionsStatisticService.PersonalSettings;
 [Route("api/personal-settings")]
 public class PersonalSettingsController(
     IPersonalSettingsRepository personalSettingsRepository,
-    IPlayerRepository playerRepository,
     PortraitCommandHandler commandHandler) : ControllerBase
 {
     private readonly IPersonalSettingsRepository _personalSettingsRepository = personalSettingsRepository;
-    private readonly IPlayerRepository _playerRepository = playerRepository;
     private readonly PortraitCommandHandler _commandHandler = commandHandler;
 
     [HttpGet("{battleTag}")]
     public async Task<IActionResult> GetPersonalSetting(string battleTag)
     {
         try
         {
-            var setting = await _personalSettingsRepository.Load(battleTag);
-            if (setting == null)
-            {
-                var player = await _playerRepository.LoadPlayerProfile(battleTag);
-                return Ok(new PersonalSetting(battleTag) { RaceWins = player });
-            }
-            return Ok(setting);
+            PersonalSetting setting = await _personalSettingsRepository.Load(battleTag, false);
+            return setting == null ? NotFound($"Personal settings of {battleTag} not found.") : Ok(setting);
```

Comment:
The previous code actually initialized the settings, instead you're now throwing an error.

So what is the actually remaining cold-boot path of a new player without settings?

Reply:
It depends on what you're doing first, I think. Logging into the launcher? Logging into the website?

Reply:
Because personal settings are created when GET'ing the api/personal-settings/{battletag} endpoint, and that's needed when visiting a profile for the first time (for example when you log in on the website for the first time, and you're redirected to your profile), I've added that back.

However, if there are no personal settings in the db already created, I ask identity service if the user is valid. If yes, they are created, otherwise I return a 404. This way I don't have to rebuild too much, and we still get protection from bogus personal settings being created.

ID PR: https://github.com/w3champions/identification-service/pull/48